src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -317,7 +323,14 @@ private Task RestartAsync()
         {
             if (!_shutdownScheduled && Interlocked.Exchange(ref _restartRequested, 1) == 0)
             {
-                return _scriptHostManager.RestartHostAsync();
+                if (Interlocked.Read(ref _suspensionRequestsCount) == 0)
```

Comment:
Looks like we're still processing notifications and just preventing the restart. Should we stop the notifications instead, like we had discussed?

Reply:
In case we suspend all notifications we loose all logs for the files/directories changes.

Reply:
I see. Makes sense.